src/utils/fp.cairo

```diff
@@ -0,0 +1,213 @@
+use starknet::storage_access::{StorePacking};
+use core::num::traits::{WideMul, Zero};
+use core::integer::{u512, u512_safe_div_rem_by_u256};
+
+pub const EPSILON: u256 = 0x10_u256;
+
+// 128.128
+#[derive(Drop, Copy)]
+pub struct UFixedPoint { 
+    pub(crate) value: u512
+}
+
+pub impl UFixedPointStorePacking of StorePacking<UFixedPoint, u256> {
```

Comment:
a u256 takes two slots, the fixed point number in this case is a 64.128 which is 192 bits which can fit in a single felt. would recommend not creating a type just for this unless it's specifically a FixedPoint64x128 type, but we haven't created special types elsewhere

Reply:
Rookie mistake.